src/plans/create/steps/general-information/PlanProjectField.tsx

```diff
@@ -0,0 +1,94 @@
+import { FC, useMemo } from 'react';
+import { useFormContext, useWatch } from 'react-hook-form';
+
+import ControlledFormGroup from '@components/common/ControlledFormGroup';
+import { V1beta1Provider } from '@kubev2v/types';
+import { useActiveNamespace } from '@openshift-console/dynamic-plugin-sdk';
+import { MenuToggleStatus } from '@patternfly/react-core';
+import { Namespace } from '@utils/constants';
+import { useForkliftTranslation } from '@utils/i18n';
+import { getDefaultNamespace } from '@utils/namespaces';
+
+import { GeneralFormFieldId, generalFormFieldLabels } from './constants';
+import { useProjectNameSelectOptions } from '@components/common/ProjectNameSelect';
+import { TypeaheadSelect } from '@components/common/TypeaheadSelect/TypeaheadSelect';
+
+type PlanProjectFieldProps = {
+  onSelect: (value: string) => void;
+};
+
+export const PlanProjectField: FC<PlanProjectFieldProps> = ({ onSelect }) => {
+  const { t } = useForkliftTranslation();
+  const {
+    formState: { errors },
+    setValue,
+  } = useFormContext();
+  const [planProject, targetProject, targetProvider, sourceProvider]: [
+    string,
+    string,
+    V1beta1Provider,
+    V1beta1Provider,
+  ] = useWatch({
+    name: [
+      GeneralFormFieldId.PlanProject,
+      GeneralFormFieldId.TargetProject,
+      GeneralFormFieldId.TargetProvider,
+      GeneralFormFieldId.SourceProvider,
+    ],
+  });
+  const [activeNamespace] = useActiveNamespace();
+  const defaultNamespace = getDefaultNamespace();
+  const projectOptions = useProjectNameSelectOptions();
+
+  const initialProjectName = useMemo(
+    () =>
+      planProject ||
+      (activeNamespace === Namespace.AllProjects ? defaultNamespace : activeNamespace),
+    [defaultNamespace, activeNamespace, planProject],
+  );
+
+  return (
+    <ControlledFormGroup
+      isRequired
+      fieldId={GeneralFormFieldId.PlanProject}
+      label={generalFormFieldLabels[GeneralFormFieldId.PlanProject]}
+      controller={{
+        ...(projectOptions?.find((option) => option.value === initialProjectName) && {
+          defaultValue: initialProjectName,
```

Comment:
why not setting the default value on context? then u don't need the check of planProject ||